ingest-app/src/cmr/ingest/services/granule_bulk_update/size/echo10.clj

```diff
@@ -0,0 +1,143 @@
+   [clojure.string :as string]
+   [clojure.zip :as zip]
+   [cmr.common.services.errors :as errors]
+   [cmr.common.xml :as cx]))
+
+(def ^:private tags-after-data-granule
+  "Defines the element tags that come after DataGranule in ECHO10 Granule xml schema"
+  #{:PGEVersionClass :Temporal :Spatial :OrbitCalculatedSpatialDomains :MeasuredParameters
+    :Platforms :Campaigns :AdditionalAttributes :InputGranules :TwoDCoordinateSystem :Price
+    :OnlineAccessURLs :OnlineResources :Orderable :DataFormat :Visible :CloudCover
+    :MetadataStandardName :MetadataStandardVersion :AssociatedBrowseImages :AssociatedBrowseImageUrls})
+
+(def ^:private tags-after-size-in-bytes
+  "Defines the element tags that come after size in a DataGranule element"
+  #{:SizeMBDataGranule :Checksum :ReprocessingPlanned :ReprocessingActual :ProducerGranuleId :DayNightFlag :ProductionDateTime
+    :LocalVersionId :AdditionalFile})
+
+(def ^:private tags-after-size-in-mb
+  "Defines the element tags that come after size in a DataGranule element"
+  #{:Checksum :ReprocessingPlanned :ReprocessingActual :ProducerGranuleId :DayNightFlag :ProductionDateTime
+    :LocalVersionId :AdditionalFile})
+
+(defn- no-data-granule-error
+  "Throws an error indicating the size parent element (DataGranule) is missing"
+  []
+  (errors/throw-service-errors :invalid-data
+   ["Can't update Size: no parent <DataGranule> element"]))
+
+(defn- get-sizes
+  "Returns a map containing the MB value and the byte value for a granule's input"
+  [size]
+  (let [sizes (map string/trim (string/split size #","))
+        mb-vals (remove nil? (map #(re-find #"^[0-9]+\.[0-9]+$" %) sizes))
+        byte-vals (remove nil? (map #(re-find #"^[0-9]+$" %) sizes))
+        size-map (merge (when (seq mb-vals) {:SizeMBDataGranule (first mb-vals)})
+                        (when (seq byte-vals) {:DataGranuleSizeInBytes (first byte-vals)}))]
+    (when (or (< 2 (count sizes)) ;;too many values
+              (< 1 (count mb-vals)) ;;too many MB values
+              (< 1 (count byte-vals)) ;;too many byte values
+              (not= (+ (count byte-vals) (count mb-vals))
+                    (count sizes)) ;;extraneous values (don't match int or double) specified
+              (not (or (seq mb-vals) (seq byte-vals)))) ;;no valid values specified
+      (errors/throw-service-errors :invalid-data
+       [(str "Can't update Size: invalid data specified. Please include at most one value for "
+             "DataGranuleSizeInBytes, and one value for SizeMBDataGranule, seperated by a comma.")]))
+    size-map))
+
+(defn- size-in-mb-element
+  "Returns an XML element for size in MB"
+  [val]
+  (xml/element :SizeMBDataGranule {} val))
+
+(defn- update-size-in-mb
+  "Takes a loc of the leftmost child loc of the DataGranule element, adds/updates MB value before returning root"
+  [start-loc sizes]
+  (let [{:keys [SizeMBDataGranule]} sizes]
+    (loop [loc start-loc done (if SizeMBDataGranule false true)]
```

Comment:
`(boolean SizeMBDataGranule)` seems like it should work here, but it doesn't play nice with `loop` for whatever reason:
```clojure had: java.lang.Boolean, needed: boolean```